recursion/src/air/ext_alu_air.rs

```diff
@@ -0,0 +1,543 @@
+use std::borrow::{Borrow, BorrowMut};
+use std::ops::{Add, Mul, Sub};
+
+use itertools::izip;
+use p3_air::{Air, AirBuilder, BaseAir};
+use p3_field::extension::BinomiallyExtendable;
+use p3_field::{Algebra, Field};
+use p3_matrix::Matrix;
+use p3_matrix::dense::RowMajorMatrix;
+use rand::distr::{Distribution, StandardUniform};
+use rand::rngs::SmallRng;
+use rand::{Rng, SeedableRng};
+
+use crate::air::alu::cols::FieldOpEvent;
+
+const D: usize = 4;
+
+/// A binomial extension element represented over a generic type `T`.
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct BinomialExtension<T>(pub [T; D]);
+
+impl<F> BinomialExtension<F> {
+    /// Creates a new binomial extension element from a base element.
+    pub fn from_base(b: F) -> Self
+    where
+        F: Field,
+    {
+        let mut arr: [F; D] = core::array::from_fn(|_| F::ZERO);
+        arr[0] = b;
+        Self(arr)
+    }
+
+    /// Returns a reference to the underlying slice.
+    pub const fn as_base_slice(&self) -> &[F] {
+        &self.0
+    }
+
+    /// Creates a new binomial extension element from a binomial extension element.
+    pub fn from<S: Into<F> + Clone>(from: &BinomialExtension<S>) -> Self {
+        BinomialExtension(core::array::from_fn(|i| from.0[i].clone().into()))
+    }
+}
+
+impl<F> Borrow<BinomialExtension<F>> for [F] {
+    fn borrow(&self) -> &BinomialExtension<F> {
+        debug_assert_eq!(self.len(), D);
+        let (prefix, shorts, _suffix) = unsafe { self.align_to::<BinomialExtension<F>>() };
+        debug_assert!(prefix.is_empty(), "Alignment should match");
+        debug_assert_eq!(shorts.len(), 1);
+        &shorts[0]
+    }
+}
+
+impl<F> BorrowMut<BinomialExtension<F>> for [F] {
+    fn borrow_mut(&mut self) -> &mut BinomialExtension<F> {
+        debug_assert_eq!(self.len(), D);
+        let (prefix, shorts, _suffix) = unsafe { self.align_to_mut::<BinomialExtension<F>>() };
+        debug_assert!(prefix.is_empty(), "Alignment should match");
+        debug_assert_eq!(shorts.len(), 1);
+        &mut shorts[0]
+    }
+}
+
+impl<T: Add<Output = T> + Clone> Add for BinomialExtension<T> {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Self(core::array::from_fn(|i| {
+            self.0[i].clone() + rhs.0[i].clone()
+        }))
+    }
+}
+
+impl<T: Sub<Output = T> + Clone> Sub for BinomialExtension<T> {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        Self(core::array::from_fn(|i| {
+            self.0[i].clone() - rhs.0[i].clone()
+        }))
+    }
+}
+
+impl<F: Add<Output = F> + Mul<Output = F> + Algebra<F>> Mul for BinomialExtension<F> {
+    type Output = Self;
+
+    fn mul(self, rhs: Self) -> Self::Output {
+        let mut result = [F::ZERO, F::ZERO, F::ZERO, F::ZERO];
+        // This value is specific for BabyBear prime's extension `F_p[x]/(x^4 - 11)`.
+        let w = F::from_u8(11); // TODO: Is this correct?
```

Comment:
same comment as in #50, I don't think we should tie the current implem to BabyBear (esp. not inside the `mul` definition itself), but rather have generic 32-bit field support from the start. This wouldn't be as cumbersome as adding Goldilocks support for instance, while directly bringing support for any 32-bit field we want (esp. KoalaBear).